jwst/dq_init/tests/test_dq_init.py

```diff
@@ -0,0 +1,340 @@
+    ref_data.meta.subarray.ysize = fullysize
+
+    # run correction step
+    outfile = do_dqinit(im, ref_data)
+
+    # read dq array
+    outpixdq = outfile.pixeldq
+
+    # check for dq flag in pixeldq of subarray image
+    assert(outpixdq[76, 100] == 1024)
+    assert(outpixdq[84, 100] == 1)
+    assert(outpixdq[114, 80] == 2048)  # check that pixel was flagged 'NO_SAT_CHECK'
+
+
+def test_dq_add1_groupdq():
+    """
+    Test if the dq_init code set the groupdq flag on the first
+    group to 'do_not_use' by adding 1 to the flag, not overwriting to 1
+    Also test whether two flags on the same pixel are added together.
+    """
+
+    # size of integration
+    nints = 1
+    ngroups = 5
+    xsize = 1032
+    ysize = 1024
+
+    # create raw input data for step
+    dm_ramp = make_rampmodel(nints, ngroups, ysize, xsize)
+
+    # create a MaskModel for the dq input mask
+
+    dq, dq_def = make_maskmodel(ysize, xsize)
+
+    # write reference file with known bad pixel values
+
+    dq[505, 505] = 1   # Do_not_use
+    dq[400, 500] = 3  # do_not_use and dead pixel
+
+    # write mask model
+    ref_data = MaskModel(dq=dq, dq_def=dq_def)
+    ref_data.meta.instrument.name = 'MIRI'
+    ref_data.meta.subarray.xstart = 1
+    ref_data.meta.subarray.xsize = xsize
+    ref_data.meta.subarray.ystart = 1
+    ref_data.meta.subarray.ysize = ysize
+
+    # set a flag in the pixel dq
+    dm_ramp.pixeldq[505, 505] = 4
+
+    # run correction step
+    outfile = do_dqinit(dm_ramp, ref_data)
+
+    # test if pixels in pixeldq were incremented in value by 1
+    assert(outfile.pixeldq[505, 505] == 5)  # check that previous dq flag is added to mask value
+    assert(outfile.pixeldq[400, 500] == 1025)  # check two flags propagate correctly
+
+
+# Set parameters for multiple runs of guider data
+args = "xstart, ystart, xsize, ysize, nints, ngroups, instrument, exp_type, detector"
+test_data = [(1, 1, 2048, 2048, 2, 2, 'FGS', 'FGS_ID-IMAGE', 'GUIDER1'),
+             (1, 1, 1032, 1024, 1, 5, 'MIRI', 'MIR_IMAGE', 'MIRIMAGE')]
+ids = ["GuiderRawModel-Image", "MIRIRampModel"]
+
+
+@pytest.mark.parametrize(args, test_data, ids=ids)
+def test_fullstep(xstart, ystart, xsize, ysize, nints, ngroups, instrument, exp_type, detector):
+    # Test that the full step runs
+
+    # create raw input data for step
+    dm_ramp = make_rawramp(instrument, nints, ngroups, ysize, xsize, xstart, ystart, exp_type)
+
+    dm_ramp.meta.instrument.name = instrument
+    dm_ramp.meta.instrument.detector = detector
+    dm_ramp.meta.observation.date = '2016-06-01'
+    dm_ramp.meta.observation.time = '00:00:00'
+
+    # run the full step
+    outfile = DQInitStep.call(dm_ramp)
+
+    # test that a pixeldq frame has been initlialized
+    if instrument == "FGS":
+        assert outfile.dq.ndim == 2
+    else:
+        assert outfile.pixeldq.ndim == 2  # a 2-d pixeldq frame exists
+
+
+def make_rawramp(instrument, nints, ngroups, ysize, xsize, xstart, ystart, exp_type=None):
```

Comment:
Seems a little confusing to me to have the arguments in the order of "ysize, xsize" and then "xstart, ystart", but as long as you know that and populate them consistently within the tests, it'll work OK.

Reply:
That came about because of two programmers working on the same code.

Reply:
The order for parameters in make_rawramp have been updated to be y first and then x.